### DISCUSSION TITLE
Different versions of Google Chrome on the same device get different device ID

### DISCUSSION THREAD

**Discussion by fyjiayou** [fingerprintjs/fingerprintjs#997] [Q&A] [answered]
![image-20240407152616176](https://github.com/fingerprintjs/fingerprintjs/assets/104678550/c86870e8-d464-45aa-a66b-b7466f73e17e)
使用的是pro版本

**Comment by makma** [comment]
@fyjiayou can you pls provide steps to reproduce?

**Reply by fyjiayou** [reply]
I installed different versions of Google Chrome on one computer, using the commercial version, but the fingerprints generated were different.

**Reply by JuroUhlar** [reply, marked as answer]
I think that this is expected to some extent. 

* When you just use your browser naturally it upgrades itself gradually in small increments. The changes between incremental versions are likely very small. That allows Fingerprint Pro to recognize it as the same browser and keep the visitor ID stable. 
* When you artificially install distant versions of the same browser and run them side by side, they will likely be recognized as separate browsers (the browser APIs Fingerprint uses to gather entropy will return sufficiently different values).

On the web platform, visitor ID identifies a browser, not a device. Chrome 98 and Chrome 123 and functionally different browsers in that regard.
